### ISSUE TITLE
accept `underscore` in enum declaration

### ISSUE THREAD

**Issue by cvng** [denoland/deno_lint#731]
### Lint Name

camelcase

### Code Snippet

```ts
enum MyEnum {
  Idenfitier1,
  Idenfitier_2,
}
```

### Expected Result

No warning

### Actual Result

Identifier 'Identifier_2' is not in camel case.

### Additional Info

<img width="522" alt="Capture d’écran 2021-06-14 à 17 22 11" src="https://user-images.githubusercontent.com/7042907/121899122-2903ae80-cd35-11eb-9fd4-638a493a6982.png">

(those are generated from a graphql schema)

### Version

deno 1.11.0 (release, x86_64-apple-darwin)
v8 9.1.269.35
typescript 4.3.2

**Comment by magurotuna** [comment]
Currently, `camelcase` rule allows PascalCase or CAPITAL_SNAKE_CASE in enum variants, so `Identifier_2` in your example should be `Identifier2` or `IDENTIFIER_2`.
Is it impossible to change your GraphQL schema to follow this naming convention for some reason? If it's possible, please consider changing the schema. Otherwise, we have to discuss whether we should allow PascalCase to include underscores. (Personally I'm in favor of it though)

**Comment by Conaclos** [comment]
@magurotuna made a relevant point.

@cvng do you agree? Can we close this issue?
